test/unit/metrics/registry.spec.js

```diff
@@ -111,6 +111,21 @@ describe("Test Metric Registry", () => {
 
 	});
 
+	describe("Test stop method", () => {
+		it("should stop reporters", () => {
+			const broker = new ServiceBroker({ logger: false });
+			const metric = new MetricRegistry(broker, {
+				collectProcessMetrics: true,
+				reporter: "Event"
+			});
+
+			metric.init();
+
+			expect(metric.stop()).toBeInstanceOf(broker.Promise);
+		});
```

Comment:
You have to check that the reporter `stop` method is called.
`metric.reporters[0].stop = jest.fn(() => Promise.resolve())`
And after stop executed
`expect(metric.reporters[0].stop).toHaveBeenCalledTimes(1);`

Reply:
Nice catch. I've added